primers.go

```diff
@@ -103,3 +108,168 @@ func MeltingTemp(sequence string) float64 {
+	}
+
+	// Index the "real" start and end position given the sequenceStart within the sequenceContext.
+	// Note: indexing of positions in Mutation notation begins at 1, so 1 is subtracted from the start
+	// and end positions
+	indexedStartPosition := sequenceStart + (mutation.StartPosition - 1)
+	indexedEndPosition := sequenceStart + (mutation.EndPosition - 1)
+
+	// Generate starter primers, in which nucleotide mutations can be added
+	seedPrimerFor, _ := MakePrimers(sequenceContext[indexedEndPosition:], targetTm)
+	_, seedPrimerRev := MakePrimers(sequenceContext[:indexedStartPosition], targetTm)
+
+	// To assemble properly, the fragments resulting from a PCR with these primers need a certain amount of overlap.
+	// This overlap will be the length of the mutation + a little homologous sequence on both ends.
+	var additionalForwardOverlap string
+	var additionalReverseOverlap string
+	additionalOverlap := overlap - len(mutation.MutatedNucleotides)
+	if additionalOverlap > 1 {
+		// Split overlap between forward and reverse
+		targetAdditionalOverlap := additionalOverlap / 2
+		additionalForwardOverlap = sequenceContext[indexedStartPosition-targetAdditionalOverlap : indexedStartPosition]
+		additionalReverseOverlap = ReverseComplement(sequenceContext[indexedEndPosition : indexedEndPosition+targetAdditionalOverlap])
+	}
+
+	// Generate final primers
+	primerFor := additionalForwardOverlap + mutation.MutatedNucleotides + seedPrimerFor
+	primerRev := additionalReverseOverlap + ReverseComplement(mutation.MutatedNucleotides) + seedPrimerRev
+
+	return primerFor, primerRev, nil
+}
+
+// Pcr simulates a PCR reaction in both the forward and reverse direction
+func Pcr(sequence string, circular bool, targetTm float64, forward string, reverse string) []string {
+	// Setup forward and reverse indexes for this PCR reaction
+	indexForward := suffixarray.New([]byte(sequence))
+	sequenceReverse := ReverseComplement(sequence)
+	indexReverse := suffixarray.New([]byte(sequenceReverse))
+
+	// Run the PCR reaction with PcrSa
+	return PcrSa(sequence, indexForward, sequenceReverse, indexReverse, circular, targetTm, forward, reverse)
+}
+
+// PcrSa simulates a PCR reaction, given suffix arrays of the forward sequence and reverse sequence of a given DNA molecule
+func PcrSa(sequenceForward string, indexForward *suffixarray.Index, sequenceReverse string, indexReverse *suffixarray.Index, circular bool, targetTm float64, forward string, reverse string) []string {
+	return append(PcrSaUnidirectional(sequenceForward, indexForward, circular, targetTm, forward, reverse), PcrSaUnidirectional(sequenceReverse, indexReverse, circular, targetTm, forward, reverse)...)
+}
+
+// PcrSaUnidirectional simulates a PCR reaction with a prebuilt suffix array (https://golang.org/pkg/index/suffixarray/) in a single direction
+func PcrSaUnidirectional(sequence string, sequenceIndex *suffixarray.Index, circular bool, targetTm float64, forward string, reverse string) []string {
```

Comment:
Code complexity is high but at first glance it looks well commented and named. 6 args again gives me pause. I'll review and think about how this can be simplified since I'm a little afraid to export this for general users to use without it being a method on `Sequence` methods.

Reply:
PcrSaUnidirectional could probably be an internal function, since it is mainly there to be used by PcrSa (PcrSa and Pcr should both be there, though)